instrumentation/redis/lib/opentelemetry/instrumentation/redis/instrumentation.rb

```diff
@@ -7,6 +7,73 @@
 module OpenTelemetry
   module Instrumentation
     module Redis
+      # Adapter for redis-client instrumentation interface
+      module ClientInstrumentation
```

Comment:
Lets move this module into its own file, either `patches/redis_client` or I guess `middlewares/redis_client` might be more appropriate here as well because its not a patch anymore.